### ISSUE TITLE
When downloading an xlsx file from a SP resultset call from a SQL question, data shown is different from the file.

### ISSUE THREAD

**Issue by FrankOsn** [metabase/metabase#13079]
**Describe the bug**
The data shown on the result set doesn't match with the data downloaded from the XLSX file option. (Correct Data).
Looks like the data shown is truncated or something.

The SP running from MSSQL is working fine. 

Note: I'm using table variable to get the grouping from another SP result. When I skip this step, the whole data result shows correctly, > 2000 rows, and no problem in showing the top 2000 on Metabase question result.

**Expected behavior**
The results on screen should match with the correct ones on the excel file.

**Screenshots**
![image](https://user-images.githubusercontent.com/9128023/89591845-d736f180-d810-11ea-85f4-faf84fdcb2c4.png)
![image](https://user-images.githubusercontent.com/9128023/89591855-df8f2c80-d810-11ea-967f-8f885051ecef.png)

**Severity**
How severe an issue is this bug to you? Is this annoying, blocking some users, blocking an upgrade or blocking your usage of Metabase entirely?
Note: the more honest and specific you are here the more we will take you seriously.
Normal, just encounter this situation.

**Metabase Diagnostic Info**

```json
{
  "browser-info": {
    "language": "en",
    "platform": "Win32",
    "userAgent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/84.0.4147.105 Safari/537.36",
    "vendor": "Google Inc."
  },
  "system-info": {
    "file.encoding": "Cp1252",
    "java.runtime.name": "Java(TM) SE Runtime Environment",
    "java.runtime.version": "1.8.0_251-b08",
    "java.vendor": "Oracle Corporation",
    "java.vendor.url": "http://java.oracle.com/",
    "java.version": "1.8.0_251",
    "java.vm.name": "Java HotSpot(TM) Client VM",
    "java.vm.version": "25.251-b08",
    "os.name": "Windows Server 2016",
    "os.version": "10.0",
    "user.language": "en",
    "user.timezone": "America/Mexico_City"
  },
  "metabase-info": {
    "databases": [
      "h2",
      "sqlserver"
    ],
    "hosting-env": "unknown",
    "application-database": "mysql",
    "application-database-details": {
      "database": {
        "name": "MariaDB",
        "version": "10.4.12-MariaDB"
      },
      "jdbc-driver": {
        "name": "MariaDB Connector/J",
        "version": "2.5.1"
      }
    },
    "run-mode": "prod",
    "version": {
      "tag": "v0.36.3",
      "date": "2020-08-04",
      "branch": "release-0.36.x",
      "hash": "a792f14"
    },
    "settings": {
      "report-timezone": null
    }
  }
}
```

**Comment by flamber** [comment]
Hi @FrankOsn 
I don't understand completely what the problem is. Can you rephrase and provide the query as well?

**Comment by FrankOsn** [comment]
Thanks @flamber. The bug is the displayed data on screen doesn't match with the data from the xlsx file when downloading.
It shows less rows on Metabase browser result than the total and correct ammounts shown on the xlsx file. 

The Metabase query is simple.
![image](https://user-images.githubusercontent.com/9128023/89655165-4d7a3900-d88f-11ea-85fc-a91e4bc0944d.png)

The image attached shows correctly the call of the SP that will return the data. Inside it, the only outstanding part is that it calls another SP to gather the data, pass the total results like 30K rows, to a mssql server table variable and do some grouping. The only agg functions in the SP are a distinct count and a sum. 

If I take out the agg functions and grouping from the table variable that stores the whole data, it runs smoothly. For presenting propuses, I know that Metabase will only show 2K rows of data, but when I proceed to download the xlxs file, all the data is in there.

I've tried deleting my cache, going on incognito mode, and even switch browsers, but same result.
I'd reviewed the issue you mention, but the rowcount isn't just the issue, on the first row, the columns are ok, but in the correct result set, I've different values. 

Let me know if more clarification is needed.
F

**Comment by flamber** [comment]
@FrankOsn I still don't understand. Metabase will only show 2000 rows unaggregated (and SQL), 10,000 rows aggregated, and allow a download 1 million rows.
I'm fairly sure it's caused by #9940, since that will limit the amount of rows the sub-queries can do, which will give you different results. Not entirely sure why it's working correctly, when you download, but I guess it's because the limit changes from 2,000 to 1 million rows.

**Comment by flamber** [comment]
Closing, since I'm fairly sure this is #9940 - if you disagree, then please provide steps to reproduce.
